### PR TITLE
Create KeyCertBundle

### PR DESCRIPTION
**Pull request by myidpt**
Use a KeyCertBundle to store the CA's key/certs in it, so that when the CA key/certs are rotated (in integrated mode), the CA is able to get the new key/certs seamlessly and consistently.

### CODE REVIEW

File: security/pkg/pki/util/testdata/3level/certs.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# This script generates all keys and certs in the 3level directory.
```
Comment (wattli) [inline security/pkg/pki/util/testdata/3level/certs.sh]: Can we rename the directory of this file, i.e., 3level to something more meaningful? And avoid having a leading number '3'.

**Comment by wattli** [comment]
Generally lgtm, but just curious, it seems this PR combine several key/cert into a structure, but how to achieve "CA is able to get the new key/certs seamlessly and consistently."? 

/lgtm

**Comment by myidpt** [comment]
It's mostly about the consistency. I'm using RWMutex to protect the access to the KeyCertBundle. Also, all values are read altogether through GetAll() or GetAllPem(), this avoids reading different cert/key out while another thread is doing the update. Seamlessly means the CA doesn't need to maintain Mutex for access control :)
Does this make sense?

**Comment by wattli** [comment]
It makes more sense now, thanks for the clarification.

**Comment by myidpt** [comment]
Great, thank you!

File: security/pkg/pki/util/testdata/3level/certs.sh (hunk 1 above)
Comment (myidpt) [reply security/pkg/pki/util/testdata/3level/certs.sh]: Sure.

**Pull request merged**